src/utils/statisticUtils.ts

```diff
@@ -95,3 +95,29 @@ export const calculateMaxDomain = (dataMax: number): number => {
 
   return Math.ceil(dataMax / 10) * 10;
 };
+
+/**
+ * A utility function to compute the approximate width of a text element
+ * rendered on an SVG canvas.
+ *
+ * @see https://stackoverflow.com/questions/118241/calculate-text-width-with-javascript/21015393#21015393
+ * @param text The text to measure
+ * @param font The font descriptor for the text element
+ * @returns The computed width of the text element or 0 if the width is not available
+ */
+export const calculateTextWidth = (text, font = "normal 11px Arial") => {
```

Comment:
Could we take the font weight, font size, and font family as separate params? Having a single param makes it easy to mess it up and it won't be immediately clear if it is used wrong.

Reply:
Definitely a good QoL suggestion. Added in 09a6ab7